### DISCUSSION TITLE
Filtering by nested object fields when lazy='joined' is set in relationship

### DISCUSSION THREAD

**Discussion by darkdkl** [sqlalchemy/sqlalchemy#8269] [Usage Questions] [answered]
Hi all, I use SQLAlchemy 1.4, I can't solve the problem

```
class ContactType(Base):
    __tablename__ = 'contact_type'
    name = Column(String(255), nullable=False)


class Contact(Base):
    __tablename__ = 'contact'

    first_name = Column(String(255), nullable=False)
    last_name = Column(String(255), nullable=False)   
    contact_type_id = Column(ForeignKey('contact_type.id'), nullable=False)
    contact_type = relationship('ContactType', lazy='joined', innerjoin=True)
```

_ideally, I would filter by Contact.contact_type.name , but it doesn't work that way_

`query = select(Contact).where(ContactType.name == 'some_type') - doesn't work`

`query = select(Contact).join(ContactType).where(ContactType.name == 'some_type')` - works, but since _contact_type = relationship('ContractorType', lazy='joined', innerjoin=True)_, it already makes a join (in other cases this feature is used) an additional join of the ContactType table is dubbed JOIN (visible when echo= True)

if I use `contains_eager`

`query = select(Contact).options(contains_eager(Contact.contact_type)).where(ContactType.name == 'some_type')` - works, but (_SAWarning: SELECT statement has a cartesian product between FROM element(s) 'contact_type. Apply join condition(s) between each element to resolve_.)

Please tell me how I can do this

**Comment by zzzeek** [comment, marked as answer]
you need to use join() and contains_eager() at the same time - join() tells it what SQL to write, contains_eager() tells the ORM how to interpret the rows:

```py
query = select(Contact).join(Contact.contact_type).options(contains_eager(Contact.contact_type)).where(ContactType.name == 'some_type')
```

**Reply by zzzeek** [reply]
docs here: https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html?highlight=contains_eager#contains-eager

**Reply by darkdkl** [reply]
Thank you, echo= True along with lazy= 'joined'
showed that JOIN already exists, I thought that it would also create a double here

**Reply by zzzeek** [reply]
the double join thing is discussed at https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#the-zen-of-joined-eager-loading
